### PR TITLE
Bugfix/do not restore virtual files

### PR DESCRIPTION
**Pull request by mgallien**

### CODE REVIEW

File: src/libsync/syncengine.cpp [hunk 1]
```diff
@@ -992,9 +992,11 @@ void SyncEngine::restoreOldFiles(SyncFileItemVector &syncItems)
             syncItem->_instruction = CSYNC_INSTRUCTION_CONFLICT;
             break;
         case CSYNC_INSTRUCTION_REMOVE:
-            qCWarning(lcEngine) << "restoreOldFiles: RESTORING" << syncItem->_file;
-            syncItem->_instruction = CSYNC_INSTRUCTION_NEW;
-            syncItem->_direction = SyncFileItem::Up;
+            if (syncItem->_type == CSyncEnums::ItemTypeFile) {
```
Comment (allexzander) [inline src/libsync/syncengine.cpp:995]: @mgallien don't we also want to restore directories `ItemTypeDirectory`?

File: test/testsynccfapi.cpp [hunk 2]
```diff
@@ -1218,6 +1218,36 @@ private slots:
...
+        fakeFolder.remoteModifier().mkdir("a/b");
+        fakeFolder.remoteModifier().insert("a/b/otherFile.txt");
+
+        //Server support finger print, but none is set.
+        fakeFolder.remoteModifier().extraDavProperties = "<oc:data-fingerprint></oc:data-fingerprint>";
+
+        fakeFolder.syncEngine().setLocalDiscoveryOptions(OCC::LocalDiscoveryStyle::DatabaseAndFilesystem);
+        QVERIFY(fakeFolder.syncOnce());
+
+        qDebug() << fakeFolder.currentLocalState();
```
Comment (allexzander) [inline test/testsynccfapi.cpp:1237]: @mgallien do you need these logs?

File: test/testsynccfapi.cpp [hunk 3]
```diff
@@ -1218,6 +1218,36 @@ private slots:
...
+        qDebug() << fakeFolder.currentRemoteState();
+
+        fakeFolder.remoteModifier().remove("a/b/otherFile.txt");
+        fakeFolder.remoteModifier().extraDavProperties = "<oc:data-fingerprint>initial_finger_print</oc:data-fingerprint>";
+
+        fakeFolder.syncEngine().setLocalDiscoveryOptions(OCC::LocalDiscoveryStyle::DatabaseAndFilesystem);
+        QVERIFY(fakeFolder.syncOnce());
+
+        qDebug() << fakeFolder.currentLocalState();
```
Comment (allexzander) [inline test/testsynccfapi.cpp:1246]: @mgallien do you need these logs?

File: test/testsynccfapi.cpp (hunk 2 above)
Comment (mgallien) [reply test/testsynccfapi.cpp:1237]: no
indeed

File: src/libsync/syncengine.cpp (hunk 1 above)
Comment (mgallien) [reply src/libsync/syncengine.cpp]: I updated the logic to allow restoration of folders
thanks

**Review by claucambra — APPROVED** [review]

**Review by allexzander — APPROVED** [review]

**Pull request merged**